script.reddit.reader/resources/lib/autoplay.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+import xbmc
+import xbmcgui
+import json
+import threading
+import time
+from Queue import Queue, Empty
+from utils import log
+
+
+q = Queue()
+def autoPlay(url, name, type_):
+    from domains import sitesBase, parse_reddit_link, ydtl_get_playable_url
+    from utils import unescape, post_is_filtered_out, strip_emoji,xbmc_busy, log, translation
+    from reddit import reddit_request, determine_if_video_media_from_reddit_json
+    from actions import setting_gif_repeat_count
+
+
+    gif_repeat_count=setting_gif_repeat_count()
+    entries = []
+    watchdog_counter=0
+    playlist = xbmc.PlayList(xbmc.PLAYLIST_VIDEO)
+    playlist.clear()
+
+    xbmc_busy()
+
+    content = reddit_request(url)
+    if not content: return
+
+    content = json.loads(content)
+
+    log("Autoplay %s - Parsing %d items" %( type_, len(content['data']['children']) )    )
+
+    for j_entry in content['data']['children']:
+        try:
+            title=unescape(j_entry['data']['title'].encode('utf-8'))
+            title=strip_emoji(title)
+
+            try:
+                media_url = j_entry['data']['url']
+            except (AttributeError,TypeError,ValueError):
+                media_url = j_entry['data']['media']['oembed']['url']
+
+            is_a_video = determine_if_video_media_from_reddit_json(j_entry)
+
+            log("  %cTITLE:%s"  %( ("v" if is_a_video else " "), title  ) )
+
+            ld=parse_reddit_link(link_url=media_url, assume_is_video=False, needs_preview=False, get_playable_url=True )
+
+            if ld:
+                log('      type:%s %s' %( ld.media_type, ld.link_action)   )
+                if ld.media_type in [sitesBase.TYPE_VIDEO, sitesBase.TYPE_GIF, sitesBase.TYPE_VIDS, sitesBase.TYPE_MIXED]:
+
+                    if ld.media_type==sitesBase.TYPE_GIF:
+                        entries.append([title,ld.playable_url, sitesBase.DI_ACTION_PLAYABLE])
+                        for _ in range( 0, gif_repeat_count ):
+                            entries.append([title,ld.playable_url, sitesBase.DI_ACTION_PLAYABLE])
+                    else:
+                        entries.append([title,ld.playable_url, ld.link_action])
+            else:
+
+                playable_video_url=ydtl_get_playable_url(media_url)
+                if playable_video_url:
+                    for u in playable_video_url:
+                        entries.append([title, u, sitesBase.DI_ACTION_PLAYABLE])
+
+        except Exception as e:
+            log( '  autoPlay exception:' + str(e) )
+
+
+    for i,e in enumerate(entries):
+        try:
+            log('  possible playable items(%.2d) %s...%s (%s)' %(i, e[0].ljust(15)[:15], e[1],e[2]) )
+        except:
+            continue
+
+    if len(entries)==0:
+        log('  Play All: no playable items' )
+        xbmc.executebuiltin('XBMC.Notification("%s","%s")' %(translation(32054), translation(32055)  ) )  #Play All     No playable items
```

Comment:
please always use the python api's instead of builtin functions. You can use here: xbmcgui.Dialog().notification()